### ISSUE TITLE
Segfault (but I don't know where from)

### ISSUE THREAD

**Issue by Tyruiop** [JuliaLang/julia#10992]
Hi!

I've tried to clean the code a lot but here is it: http://tyruiop.eu/~tyruiop/debug.tar . The ImTransform folder is a small Package I made for myself which is needed to reproduce the bug.

Also, I need to run the "testemloop.jl" script three times to have the segfault inside julia 0.3.7

And here is what gdb tells me (after gdb --args julia-debug ./testemloop.jl):

```
ERROR: 
 in - at array.jl:723
 in emloop at /home/tyruiop/PhD/Julia/testemloop.jl:51
 in include at /usr/bin/../lib/julia/sys.so
 in include_from_node1 at loading.jl:128
 in process_options at /usr/bin/../lib/julia/sys.so
 in _start at /usr/bin/../lib/julia/sys.sofatal: error thrown and no exception handler available.
Base.TypeError(func=:Gray, context="T", expected=Void<:T<:Union(FixedPointNumbers.FixedPoint, FloatingPoint), got=Symbol)
```

And what happens when I include for the third time testemloop.jl:

```
julia> include("testemloop.jl")

signal (11): Erreur de segmentation
unknown function (ip: -1081875251)
unknown function (ip: -1081874158)
jl_gc_collect at /usr/bin/../lib/julia/libjulia.so (unknown line)
allocobj at /usr/bin/../lib/julia/libjulia.so (unknown line)
jl_alloc_tuple_uninit at /usr/bin/../lib/julia/libjulia.so (unknown line)
jl_alloc_tuple at /usr/bin/../lib/julia/libjulia.so (unknown line)
jlcall_size_21166 at  (unknown line)
jl_apply_generic at /usr/bin/../lib/julia/libjulia.so (unknown line)
calcTheta at /home/tyruiop/PhD/Julia/testemloop.jl:23
jl_apply_generic at /usr/bin/../lib/julia/libjulia.so (unknown line)
emloop at /home/tyruiop/PhD/Julia/testemloop.jl:46
jlcall_emloop_21299 at  (unknown line)
jl_apply_generic at /usr/bin/../lib/julia/libjulia.so (unknown line)
unknown function (ip: -1081963992)
unknown function (ip: -1081967463)
unknown function (ip: -1081898787)
unknown function (ip: -1081898649)
unknown function (ip: -1081896080)
jl_load at /usr/bin/../lib/julia/libjulia.so (unknown line)
include at /usr/bin/../lib/julia/sys.so (unknown line)
jl_apply_generic at /usr/bin/../lib/julia/libjulia.so (unknown line)
include_from_node1 at /usr/bin/../lib/julia/sys.so (unknown line)
jl_apply_generic at /usr/bin/../lib/julia/libjulia.so (unknown line)
unknown function (ip: -1081963992)
unknown function (ip: -1081967463)
unknown function (ip: -1081898787)
jl_f_top_eval at /usr/bin/../lib/julia/libjulia.so (unknown line)
eval_user_input at REPL.jl:53
jlcall_eval_user_input_20070 at  (unknown line)
jl_apply_generic at /usr/bin/../lib/julia/libjulia.so (unknown line)
anonymous at task.jl:95
jl_handle_stack_switch at /usr/bin/../lib/julia/libjulia.so (unknown line)
julia_trampoline at /usr/bin/../lib/julia/libjulia.so (unknown line)
unknown function (ip: 4199613)
__libc_start_main at /usr/bin/../lib/libc.so.6 (unknown line)
unknown function (ip: 4199673)
unknown function (ip: 0)
[1]    9697 segmentation fault (core dumped)  julia
```

**Comment by pao** [comment]
Could you please also paste the output of `versioninfo()`? Thanks.

**Comment by Tyruiop** [comment]
Yes sorry, here it is:

Julia Version 0.3.7
Commit cb9bcae (2015-03-23 21:36 UTC)
Platform Info:
  System: Linux (x86_64-unknown-linux-gnu)
  CPU: Intel(R) Core(TM) i5-4200U CPU @ 1.60GHz
  WORD_SIZE: 64
  BLAS: libopenblas (DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: libopenblas
  LIBM: libm
  LLVM: libLLVM-3.3

**Comment by jakebolewski** [comment]
I could reproduce the segfault using the version of Julia you posted, but this looks to be fixed on the latest v0.3 release branch.
